libs/market-depth/src/lib/orderbook-data.ts

```diff
@@ -286,6 +301,41 @@ export const updateCompactedRows = (
   return data;
 };
 
+/**
+ * Updates OrderbookData[] with new data received from subscription - mutates input
+ *
+ * @param rows
+ * @param sell
+ * @param buy
+ * @param resolution
+ * @returns void
+ */
+export const updateCompactedRowsByType = (
+  oldData: Readonly<OrderbookRowData[]>,
+  newData: Readonly<PriceLevelFieldsFragment[]> | null,
+  resolution: number,
+  dataType: VolumeType
+) => {
+  const data = cloneDeep(oldData as OrderbookRowData[]);
+  uniqBy(reverse(newData || []), 'price')?.forEach((delta) => {
+    partiallyUpdateCompactedRows(dataType, data, delta, resolution);
+  });
+
+  updateCumulativeVolumeByType(data, dataType);
+  let index = 0;
+  // remove levels that do not have any volume
+  while (index < data.length) {
```

Comment:
Why do we need this? We shouldn't need to fill in empty price levels. That was a previous requirement

Reply:
empty bids or asks can be a result of updating existing data with delta from subscription